### PR TITLE
Refactor popup button actions to an enum #291

### PR DESCRIPTION
**Pull request by NohaFahmi**
This PR includes fix of #291: refactoring button actions to an enum.

- [x] added 2 enums one for types corresponding to numbers (actions types) and one for types corresponds to names (actions names)
- [x] add actions options list to constants
- [x] handle using enums in popup components and update handle button Action functionality
- [x] handle sending the selected action type enum value not name to create/edit payload for APIs
- [x] handle using the same enum in backend 
- [ ] apply same on banners and hints (waiting for confirmation on what's done on popups)

### CODE REVIEW

**Comment by NohaFahmi** [comment]
Note: I couldn't test the changes I done on backend cause I had an issue running the DB, I run the docker successfully and APIs locally working fine in postman but it return 500 due to DB error.
I get this error when run the backend project and I couldn't figure a way to fix it

image

**Comment by erenfn** [comment]
> Note: I couldn't test the changes I done on backend cause I had an issue running the DB, I run the docker successfully and APIs locally working fine in postman but it return 500 due to DB error. I get this error when run the backend project and I couldn't figure a way to fix it
> 
> image

Use docker: `docker compose up` or change host in backend/config/config from db to localhost

**Comment by erenfn** [comment]
Also take a look at coderabbit's suggestions; some actually look useful

**Review by erenfn — CHANGES_REQUESTED** [review]
- [ ] Can you also modify the frontend and backend for the Banner and Hint? They are pretty similar. (You can do this in a separate PR if you prefer.)

- [ ] Additionally, can you add a functionality to the backend where, if the buttonAction is a string, it is converted into an integer using the enum?

File: frontend/src/utils/constants.js [hunk 1]
```diff
@@ -7,3 +7,18 @@ export const API_BASE_URL = 'https://onboarding-demo.bluewavelabs.ca/api/';
...
 export const APP_TITLE = 'Bluewave Onboarding';
 export const SUPPORT_EMAIL = 'support@bluewave.com';
+
+export const ACTIONS_TYPES = Object.freeze({
+    NO_ACTION: 1,
+    OPEN_URL: 2,
+    OPEN_URL_IN_NEW_TAB: 3,
+    CLOSE_POPUP: 4,
+});
+export const ACTIONS_Names = Object.freeze({
```
Comment (uparkalau) [inline frontend/src/utils/constants.js:17]: All constant name has to be CAPITAL

File: backend/src/models/Popup.js [hunk 2]
```diff
@@ -11,7 +11,7 @@ module.exports = (sequelize, DataTypes) => {
         autoIncrement: true,
       },
       closeButtonAction: {
-        type: DataTypes.STRING,
+        type: DataTypes.INTEGER,
```
Comment (uparkalau) [inline backend/src/models/Popup.js:14]: Since you are making changes in the database you have to create a migration for that

**Comment by NohaFahmi** [comment]
Hi @erenfn , @uparkalau, 
I did the changes you've requested but I am getting an error regarding importing from constants.json inside shared floder in root, here it is, it happens even if I used relative path to shared in import statement and  I can't resolve it, any advice?
image

**Comment by erenfn** [comment]
I'll have to look into that, but my guess is 'npm run dev' only uses the files under that folder (I may be wrong). Since shared it outside that folder, it may not exist in the server files. If it is that case, me may either have to change the scope or move the enums back to the folders. For this instance,  I'll prefer the second one.

**Comment by NohaFahmi** [comment]
> I'll have to look into that, but my guess is 'npm run dev' only uses the files under that folder (I may be wrong). Since shared it outside that folder, it may not exist in the server files. If it is that case, me may either have to change the scope or move the enums back to the folders. For this instance, I'll prefer the second one.

so do I go for the second option and move the enums back to folders as you suggested?

**Comment by erenfn** [comment]
> I'll have to look into that, but my guess is 'npm run dev' only uses the files under that folder (I may be wrong). Since shared it outside that folder, it may not exist in the server files. If it is that case, me may either have to change the scope or move the enums back to the folders. For this instance, I'll prefer the second one.

It turns out this wasn't the issue because I was able to run both the frontend and backend servers locally. The errors only occur in Docker. If you broaden the scope to include the shared folder, I think it will solve the issue.

**Comment by erenfn** [comment]
How is this issue going?

**Pull request closed without merging**